send_stream.go

```diff
@@ -202,10 +201,12 @@ func (s *sendStream) getDataForWriting(maxBytes protocol.ByteCount) ([]byte, boo
 
 	var ret []byte
 	if protocol.ByteCount(len(s.dataForWriting)) > maxBytes {
-		ret = s.dataForWriting[:maxBytes]
+		ret = make([]byte, int(maxBytes))
```

Comment:
It would be nice if the `make([]byte)` call can be replaced with `getPacketBuffer()`. I guess this can be done later.

Reply:
Funny, I just discussed this with @lucas-clemente half an hour ago. The complicated part about this when to return the buffer. I opened #1677 to track this.